ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -173,9 +173,12 @@
       lastModifiedTimeSec = new AtomicLong(0);
       if (shouldMap) {
         map(false);
+        bloomFile = new File(indexFile.getParent(), indexSegmentFilenamePrefix + BLOOM_FILE_NAME_SUFFIX);
+        if (!bloomFile.exists()) {
+          generateBloomFileAndPersist();
+        }
         // Load the bloom filter for this index
```

Comment:
Can't this go in the `else` block? Why read from the file and deserialize to populate the `bloomFilter` when it has already been done?

Reply:
sounds reasonable. Let me make a quick change.